clients/cli/src/orchestrator/client.rs

```diff
@@ -129,6 +136,78 @@ impl OrchestratorClient {
             verifying_key.to_bytes().to_vec(),
         )
     }
+
+    /// Detects the user's country for network optimization purposes.
+    ///
+    /// Privacy Note: This only detects the country (2-letter code like "US", "CA", "GB")
+    /// and does NOT track precise location, IP address, or any personally identifiable
+    /// information. The country information helps the Nexus network route requests to
+    /// the nearest servers for better performance and reduced latency.
+    ///
+    /// The detection is cached for the duration of the program run.
+    async fn get_country(&self) -> String {
+        if let Some(country) = COUNTRY_CODE.get() {
+            return country.clone();
+        }
+
+        let country = self.detect_country().await;
+        let _ = COUNTRY_CODE.set(country.clone());
+        country
+    }
+
+    async fn detect_country(&self) -> String {
+        // Try Cloudflare first (most reliable)
+        if let Ok(country) = self.get_country_from_cloudflare().await {
+            return country;
+        }
+
+        // Fallback to ipinfo.io
+        if let Ok(country) = self.get_country_from_ipinfo().await {
+            return country;
+        }
+
+        "US".to_string() // Final fallback
```

Comment:
The comment here should be a bit more discursive, and make clear this is only used for routing efficiency, nothing else, so misidentification causes no issue beyond latency.